solr/core/src/java/org/apache/solr/search/SolrIndexSearcher.java

```diff
@@ -197,9 +197,17 @@ private static DirectoryReader getReader(
   private static DirectoryReader wrapReader(SolrCore core, DirectoryReader reader)
       throws IOException {
     assert reader != null;
-    return ExitableDirectoryReader.wrap(
-        UninvertingReader.wrap(reader, core.getLatestSchema().getUninversionMapper()),
-        SolrQueryTimeoutImpl.getInstance());
+    switch (IndexSchema.getUninvertibleSupport()) {
```

Comment:
feels like it should be an instance method on the IndexSchema, not a global lookup.  Even if it may be toggle-able at a global level.